app/Http/Controllers/IncidentController.php

```diff
@@ -33,25 +33,54 @@ public function active(Request $request): JsonResponse
     public function search(IncidentSearchRequest $request): JsonResponse
     {
         $day = $request->get('day');
+        $before = $request->get('before');
+        $after = $request->get('after');
+
+        if($day && ($before || $after)){
+            abort(422);
+        }
+
+        $timeRange = false;
+        if($after){
+            $after = new Carbon($after);
+            $before = $before ? new Carbon($before) : new Carbon();
+            $timeRange = [$after, $before];
+        }
+
+        if($day){
+            $day = new Carbon($day);
+        }
 
-        $date = new Carbon($day);
         $all = $request->get('all');
+        $extend = $request->get('extend');
         $concelho = $request->get('concelho');
 
-        $incidents = Incident::whereBetween(
+
+        $incidents = Incident::when($day, function($query, $day){
+            return $query->whereBetween(
+                'dateTime',
+
+                [
+                    Carbon::parse($day->startOfDay()),
+                    Carbon::parse($day->endOfDay())
+                ]
+            );
+        })->when($timeRange, function($query,$timeRange){
+            return $query->whereBetween(
                 'dateTime',
                 [
-                    Carbon::parse($date->startOfDay()),
-                    Carbon::parse($date->endOfDay())
+                    Carbon::parse($timeRange[0]->startOfDay()),
+                    Carbon::parse($timeRange[1]->endOfDay())
```

Comment:
According to the Carbon docs, the `startOfDay()` and `endOfDay()` methods returns new Carbon instances. I think the parse() call is not required, right?

Reply:
Only work this way. If i remove the `parse()` it doesn't work.